### PR TITLE
Release 2.5.0

### PR DESCRIPTION
**Pull request by jonasscheid**
- Support for brukers tdf format by adding tdf2mzml converter [#263](https://github.com/nf-core/mhcquant/issues/263)
- DeepLC retention time prediction
- MS2PIP peak intensity prediction
- Added OpenMS FileFilter to clean mzml after parsing to remove artifacts like empty spectra or precursors with charge 0 (optional)
- Made file extension check case insensitive
- Added option to provide a default comet parameters file
- Optimize resource allocations
- Update to latest template 2.10 [#282](https://github.com/nf-core/mhcquant/pull/282)
- Restructured quantification to make use of enhanced rescoring:
Before
mhcquant_subway_old

After
mhcquant_subway

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mhcquant/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/mhcquant _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jonasscheid** [comment]
test_quant and test_full run through locally. I need to adjust resource allocations / time limits. But first I want to gather the comments of reviews to bundle it

File: README.md [hunk 1]
```diff
@@ -22,54 +23,95 @@ The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool
...
+## Pipeline output
+
+=======
+
+```bash
+nextflow run nf-core/mhcquant -profile test,<docker/singularity/podman/shifter/charliecloud/conda/institute> \
+                              --input 'samples.tsv' \
+                              --fasta 'SWISSPROT_2020.fasta' \
+                              --outdir ./results
+```
```
Comment (mapo9) [inline README.md:79]: should this example be in the output?

File: subworkflows/local/quant.nf [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                                    .map { meta, aligned_idxml -> [[id: meta.sample + '_' + meta.condition], aligned_idxml] }
+                                    .groupTuple())
+        ch_versions = ch_versions.mix(OPENMS_IDMERGER_QUANT.out.versions.ifEmpty(null))
+
+        // Manipulate channels such that we end up with : [meta, mzml, run_idxml, merged_runs_idxml]
+        MAP_ALIGNMENT.out.aligned_mzml
+                .join(MAP_ALIGNMENT.out.aligned_idxml)
+                .map {meta, mzml, idxml -> [[id: meta.sample + '_' + meta.condition], meta, [id:meta.id, file:mzml], [id:meta.id, file:idxml]] }
+                .groupTuple(sort: sortById)
+                .map { group_meta, meta, mzml, idxml -> [group_meta, meta, mzml.file, idxml.file]}
```
Comment (mapo9) [inline subworkflows/local/quant.nf:72]:
```suggestion
                .map { group_meta, meta, mzml, idxml -> [group_meta, meta, mzml.file, idxml.file] }
```

**Review by mapo9 — CHANGES_REQUESTED** [review]
Except for the minor cosmetic changes:
LGTM!

File: modules/local/tdf2mzml.nf [hunk 3]
```diff
@@ -0,0 +1,27 @@
+process TDF2MZML {
+    tag "$meta.id"
+    //label 'process_medium'
+
+    container "docker.io/mfreitas/tdf2mzml"
```
Comment (steffenlem) [inline modules/local/tdf2mzml.nf:5]: conda and singularity are not available here?

File: nextflow.config [hunk 4]
```diff
@@ -62,6 +67,19 @@ params {
     variable_mods                   = 'Oxidation (M)'
     vcf_sheet                       = null
     annotate_ions                   = false
+    filter_mzml                     = false
+
+    // DeepLC settings
+    use_deeplc                      = false
+    deeplc_calibration_mode         = 'rt_bin'
```
Comment (steffenlem) [inline nextflow.config:74]: We could change this to 'idx_bin' since it outperformed the 'rt_bin' calibration in most cases.

**Review by steffenlem — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -22,54 +23,95 @@ The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool
...
+Now, you can run the pipeline using:
+
+<!-- TODO nf-core: update the following command to include all required parameters for a minimal example -->
+
+```bash
+nextflow run nf-core/mhcquant \
+   -profile <docker/singularity/.../institute> \
+   --input samplesheet.csv \
+   --outdir <OUTDIR>
+```
```
[the remark this answers is not in the document]
Comment (jonasscheid) [reply README.md:60]: No these things are issues arising due to auto.merging of the new template... I'll check the affected files again 👍🏼

File: README.md (hunk 1 above)
Comment (jonasscheid) [reply README.md:79]: Same here, thanks good catch!

File: bin/check_samplesheet.py [hunk 6]
```diff
@@ -25,13 +25,15 @@ class RowChecker:
 
     """
 
-    VALID_FORMATS = (
-        ".raw",
-        ".mzML",
-    )
+    VALID_FORMATS = (".raw", ".mzml", ".d")
```
[the remark this answers is not in the document]
Comment (jonasscheid) [reply bin/check_samplesheet.py:28]: That is black formatting. Nothing I am allowed to do here, else linter test crashes :D

**Review by mapo9 — APPROVED** [review]

**Comment by mapo9** [comment]
LGTM! 🚀

**Review by steffenlem — APPROVED** [review]
Looks good!

**Pull request merged**